File: code_samples/lua/design_patterns_adapter.lua
-- Design Patterns: Adapter
-- AI/ML Training Sample

Adapter = {}
Adapter.__ index = Adapter

function Adapter:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Adapter:process(input)
    self.data = input
end

function Adapter:getData()
    return self.data
end

function Adapter:validate()
    return #self.data > 0
end

-- Example usage
local instance = Adapter:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/typescript/design_patterns_adapter.ts
/**
 * Design Patterns: Adapter
 * AI/ML Training Sample
 */

interface IAdapter {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Adapter implements IAdapter {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Adapter();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Adapter, IAdapter };
